aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -413,11 +423,17 @@ sub remove_vn_ars
 # Update VN AR leases
 sub update_vn_ars
 {
-    my ($self, $one, $vnet, $ardata, $ar) = @_;
-    my $arid;
+    my ($self, $one, $arinfo, $vnet, $ardata, $ar) = @_;
+    my ($arid, $modtpl);
     # Update the AR info
-    $main::this_app->debug(1, "AR template to update from $vnet: ", $ardata->{ar});
-    $arid = $ar->updatear($ardata->{ar});
+    $arid = $self->detect_vn_ar_quattor($arinfo);
+    if (defined($arid)) {
```

Comment:
Yes, is quite ugly, in fact we have fixed this in the component  https://github.com/quattor/configuration-modules-core/pull/734/files... I will fix this in PR #181